runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -2584,10 +2633,10 @@ done:;
 		j9object_t threadLock = J9VMJAVALANGTHREAD_LOCK(_currentThread, _currentThread->threadObject);
 		if (!VM_ObjectMonitor::inlineFastObjectMonitorEnter(_currentThread, threadLock)) {
 			IDATA monitorRC = objectMonitorEnterNonBlocking(_currentThread, threadLock);
-			if (0 == monitorRC) {
+			if (J9_OBJECT_MONITOR_OOM == monitorRC) {
 				rc = THROW_MONITOR_ALLOC_FAIL;
 				goto done;
-			} else if (1 == monitorRC) {
+			} else if (J9_OBJECT_MONITOR_BLOCKING == monitorRC) {
```

Comment:
Do you need to handle other cases here (are we 100% sure Thread will never be a VT?)

Reply:
j.l.Thread has a known subclass `ForkJoinWorkerThread` so it could never be a valuetype